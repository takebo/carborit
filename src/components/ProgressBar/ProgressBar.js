import React from 'react';
import PropTypes from 'prop-types';
import './ProgressBar.scss';

const ProgressBar = props => {
  return (
    <div className={props.complete ? 'progressbar complete' : 'progressbar'} />
  );
};

ProgressBar.propTypes = {
  complete: PropTypes.bool,
};

export default ProgressBar;
