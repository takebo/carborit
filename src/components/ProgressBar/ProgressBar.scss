.progressbar {
  background: #fcfbfb;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 5px;

  &:before {
    content: '';
    background-color: #f14444;
    position: absolute;
    width: 0;
    height: 100%;
    left: 0;
    top: 0;
    z-index: 1;
    animation: 300ms start forwards;
  }
  &:after {
    content: '';
    background-color: #f14444;
    position: absolute;
    width: 0;
    height: 100%;
    left: 0;
    top: 0;
    animation: 400ms progress forwards;
    animation-delay: 200ms;
  }
  &.complete:after {
    width: 30%;
    animation: 200ms complete forwards;
  }
}

@keyframes start {
  from {
    width: 0%;
  }
  to {
    width: 10%;
  }
}

@keyframes progress {
  from {
    width: 0%;
  }
  to {
    width: 30%;
  }
}

@keyframes complete {
  0% {
    width: 30%;
  }
  100% {
    width: 100%;
  }
}
