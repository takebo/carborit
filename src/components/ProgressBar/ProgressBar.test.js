import React from 'react';
import { shallow } from 'enzyme';

import ProgressBar from './ProgressBar';

describe('ProgressBar', () => {
  const props = { complete: false };
  const component = shallow(<ProgressBar {...props} />);
  it('should render', () => {
    expect(component).toMatchSnapshot();
  });

  it('should add class complete', () => {
    const props = { complete: true };
    const component = shallow(<ProgressBar {...props} />);
    expect(component.hasClass('complete')).toBe(true);
    expect(component).toMatchSnapshot();
  });
});
