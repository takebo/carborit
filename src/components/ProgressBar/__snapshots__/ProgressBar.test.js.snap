// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should add class complete 1`] = `
<div
  className="progressbar complete"
/>
`;

exports[`ProgressBar should render 1`] = `
<div
  className="progressbar"
/>
`;
